json - bug on Url column deserialization

JsonType mistake : 

`Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve type id 'Url' as a subtype of [simple type, class fr.ign.validator.model.AttributeType<java.lang.Object>]: known type ids = [Boolean, Date, Double, Filename, Geometry, GeometryCollection, Integer, LineString, MultiLineString, MultiPoint, MultiPolygon, Path, Point, Polygon, String, UrlType] (for POJO property 'columns')`

--- a/validator-core/src/main/java/fr/ign/validator/model/type/UrlType.java
+++ b/validator-core/src/main/java/fr/ign/validator/model/type/UrlType.java
@@ -3,6 +3,8 @@ package fr.ign.validator.model.type;
 import java.net.MalformedURLException;
 import java.net.URL;
 
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
 import fr.ign.validator.model.AttributeType;
 
 /**
@@ -10,7 +12,10 @@ import fr.ign.validator.model.AttributeType;
  * 
  * @author MBorne
  */
+@JsonTypeName(UrlType.TYPE)
 public class UrlType extends AttributeType<URL> {
+
+    public static final String TYPE = "Url";
 
     public UrlType() {
         super(URL.class);
@@ -18,7 +23,7 @@ public class UrlType extends AttributeType<URL> {
 
     @Override
     public String getTypeName() {
-        return "Url";
+        return TYPE;
     }
 
     /**

--- a/validator-core/src/test/java/fr/ign/validator/io/JsonModelReaderTest.java
+++ b/validator-core/src/test/java/fr/ign/validator/io/JsonModelReaderTest.java
@@ -130,7 +130,7 @@ public class JsonModelReaderTest {
     private void assertExceptedFeatureTypeAdresse(FeatureType featureType) {
         Assert.assertEquals("ADRESSE", featureType.getName());
         Assert.assertEquals("Table d'identifiant et adresse", featureType.getDescription());
-        Assert.assertEquals(2, featureType.getAttributeCount());
+        Assert.assertEquals(3, featureType.getAttributeCount());
         Assert.assertFalse(featureType.isSpatial());
 
         int index = 0;
@@ -151,7 +151,7 @@ public class JsonModelReaderTest {
             AttributeType<?> attribute = featureType.getAttribute(index++);
             Assert.assertEquals("ADRESSE", attribute.getName());
             Assert.assertEquals("String", attribute.getTypeName());
-            Assert.assertEquals("Une adresse dans une ville", attribute.getDescription());
+            Assert.assertEquals("Le libellé de l'adresse", attribute.getDescription());
             Assert.assertFalse(
                 attribute.getConstraints().isRequired()
             );
@@ -161,6 +161,19 @@ public class JsonModelReaderTest {
             Assert.assertEquals(
                 Integer.valueOf(254),
                 attribute.getConstraints().getMaxLength()
+            );
+        }
+
+        {
+            AttributeType<?> attribute = featureType.getAttribute(index++);
+            Assert.assertEquals("HOMEPAGE", attribute.getName());
+            Assert.assertEquals("Url", attribute.getTypeName());
+            Assert.assertEquals("Page de présentation", attribute.getDescription());
+            Assert.assertFalse(
+                attribute.getConstraints().isRequired()
+            );
+            Assert.assertFalse(
+                attribute.getConstraints().isUnique()
             );
         }
     }

--- a/validator-core/src/test/resources/config-json/adresse/types/ADRESSE.json
+++ b/validator-core/src/test/resources/config-json/adresse/types/ADRESSE.json
@@ -13,11 +13,19 @@
     },
     {
       "name": "ADRESSE",
-      "description": "Une adresse dans une ville",
+      "description": "Le libellé de l'adresse",
       "type": "String",
       "constraints": {
         "required": false,
         "maxLength": 254
+      }
+    },
+    {
+      "name": "HOMEPAGE",
+      "description": "Page de présentation",
+      "type": "Url",
+      "constraints": {
+        "required": false
       }
     }
   ]

